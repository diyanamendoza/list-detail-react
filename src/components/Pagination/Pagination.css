nav {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 20px;
    font-family: 'Source Code Pro', monospace;
    color: white;
}