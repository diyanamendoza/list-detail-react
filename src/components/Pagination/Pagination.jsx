import './Pagination.css';

export default function Pagination({currPage, viewNextPage, viewPrevPage}) {

const leftArrow = <svg onClick={viewPrevPage} cursor='pointer'
                    xmlns="http://www.w3.org/2000/svg" width='30' height='30' fill="none" viewBox="0 0 24 24" stroke="currentColor">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M11 19l-7-7 7-7m8 14l-7-7 7-7" />
                </svg>

const rightArrow = <svg onClick={viewNextPage} cursor='pointer'
                    xmlns="http://www.w3.org/2000/svg" width='30' height='30' fill="none" viewBox="0 0 24 24" stroke="currentColor">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M13 5l7 7-7 7M5 5l7 7-7 7" />
                </svg>


    return (
        <nav>
            {(currPage > 1) && leftArrow}
            <p>Page {currPage}</p>
            {(currPage < 26) && rightArrow}
        </nav>
    )
}
