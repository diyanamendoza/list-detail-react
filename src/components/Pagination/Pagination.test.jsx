import { render, screen } from '@testing-library/react'
import Pagination from './Pagination'

it('should render the pagination nav', () => {
    const { container } = render(
         <Pagination currPage={1}/>
        )
    expect(container).toMatchSnapshot()
    screen.getByText(/Page 1/i)
})