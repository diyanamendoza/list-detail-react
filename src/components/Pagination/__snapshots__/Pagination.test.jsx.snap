// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the pagination nav 1`] = `
<div>
  <nav>
    <p>
      Page 
      1
    </p>
    <svg
      cursor="pointer"
      fill="none"
      height="30"
      stroke="currentColor"
      viewBox="0 0 24 24"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 5l7 7-7 7M5 5l7 7-7 7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </nav>
</div>
`;
